tests_integration/testing_utils/file_utils.py

```diff
@@ -0,0 +1,6 @@
+def replace_text_in_file(file_path: str, to_replace: str, replacement: str) -> None:
+    with open(file_path, "r") as file:
```

Comment:
Maybe just single open with `w+`?

Reply:
I think it won't work.

* `w+` won't work because it will truncate the file (https://stackoverflow.com/a/30931305)
* `r+` also won't work because it will move the file pointer while reading
* current solution is like this one: https://stackoverflow.com/a/17141572